PulsePace/Gameplay/ViewModels/GameViewModel.swift

```diff
@@ -29,16 +35,39 @@ class GameViewModel: ObservableObject {
         50
     }
 
-//    var hitObjects: PriorityQueue<any GameHO> {
-//        PriorityQueue<any GameHO>(sortBy: <)
-//    }
+    lazy var sceneAdaptor: ([Entity: any GameHO]) -> Void = { [weak self] gameHOTable in
+        self?.clear()
+        gameHOTable.forEach { gameHOEntity in
+            if let slideGameHO = gameHOEntity.value as? SlideGameHO {
+                self?.slideGameHOs.append(SlideGameHOVM(gameHO: slideGameHO, id: gameHOEntity.key.id))
+            } else if let holdGameHO = gameHOEntity.value as? HoldGameHO {
+                self?.holdGameHOs.append(HoldGameHOVM(gameHO: holdGameHO, id: gameHOEntity.key.id))
+            } else if let tapGameHO = gameHOEntity.value as? TapGameHO {
+                self?.tapGameHOs.append(TapGameHOVM(gameHO: tapGameHO, id: gameHOEntity.key.id))
+            } else {
+                print("Unidentified game HO type")
+            }
+        }
+    }
+
+    private func clear() {
+        slideGameHOs.removeAll()
```

Comment:
```suggestion
        slideGameHOs = []
```
Can assign empty array directly